### PR TITLE
DESENG-483: Added TimeLine widget and Video Widget Unit Test

### PR DESCRIPTION
**Pull request by ratheesh-aot**
Issue #: https://apps.itsm.gov.bc.ca/jira/browse/DESENG-483

*Description of changes:*
Added TimeLine widget and Video Widget Unit Test

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

File: met-web/tests/unit/components/factory.ts [hunk 1]
```diff
@@ -198,6 +216,33 @@ const mockPoll: PollWidget = {
     answers: [mockPollAnswer1, mockPollAnswer2],
 };
 
+const mockVideo: VideoWidget = {
+    id: 1,
+    widget_id: 1,
+    engagement_id: 1,
+    video_url: 'https://www.youtube.com/watch?v=dQw4w9WgXcQ',
```
Comment (Baelx) [inline met-web/tests/unit/components/factory.ts:223]: 😂

File: met-web/tests/unit/components/widgets/TimeLineWidget.test.tsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        const descInput = document.querySelector('textarea[name="description"]') as HTMLInputElement;
+
+        fireEvent.change(titleInput, { target: { value: mockTimeLine.title } });
+        fireEvent.change(descInput, { target: { value: mockTimeLine.description } });
+
+        // Adding First Event
+        const eventDesc1 = document.querySelector('input[name="eventDescription1"]') as HTMLInputElement;
+        const eventTime1 = document.querySelector('input[name="eventTime1"]') as HTMLInputElement;
+        const selectElement1 = document.querySelector('input[name="eventStatus1"]') as HTMLInputElement;
+        console.log(selectElement1);
```
Comment (Baelx) [inline met-web/tests/unit/components/widgets/TimeLineWidget.test.tsx:120]: Is this console.log still needed?

File: met-web/tests/unit/components/widgets/VideoWidget.test.tsx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+import { EngagementSettings, createDefaultEngagementSettings } from 'models/engagement';
+
+jest.mock('components/map', () => () => <div></div>);
+
+const mockEngagementSettings: EngagementSettings = {
+    ...createDefaultEngagementSettings(),
+};
+
+jest.mock('axios');
+jest.mock('react-redux', () => ({
```
Comment (Baelx) [inline met-web/tests/unit/components/widgets/VideoWidget.test.tsx:27]: I wonder if we could set up a module that gets included for these widget tests. I noticed there's a lot of the same code declared at the top of each

**Review by Baelx — CHANGES_REQUESTED** [review]
Things look great, thanks for these! Could you please just remove the console.log or state why it's needed?

File: met-web/tests/unit/components/widgets/TimeLineWidget.test.tsx (hunk 2 above)
Comment (ratheesh-aot) [reply met-web/tests/unit/components/widgets/TimeLineWidget.test.tsx:71]: Removed

File: met-web/tests/unit/components/widgets/VideoWidget.test.tsx (hunk 3 above)
Comment (ratheesh-aot) [reply met-web/tests/unit/components/widgets/VideoWidget.test.tsx:15]: Moved most of the Jest mock and Spy codes to a outside reusable functions

File: met-web/tests/unit/components/factory.ts (hunk 1 above)
Comment (ratheesh-aot) [reply met-web/tests/unit/components/factory.ts]: Fixed

Comment (Baelx) [reply met-web/tests/unit/components/factory.ts]: Nothing needed fixing in my opinion!

**Review by Baelx — APPROVED** [review]

**Pull request merged**